src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -251,38 +377,50 @@ private SourceArchive(Stream stream)
         }
 
         private static string CreateMetadataFileContents(
-            Uri baseFolderUri,
             string entrypointPath,
             IEnumerable<SourceFileInfoEntry> files,
-            IReadOnlyDictionary<Uri, SourceCodeDocumentUriLink[]>? documentLinks
+            IReadOnlyDictionary<string, SourceCodeDocumentPathLink[]>? documentLinks
         )
         {
-            // Add the __metadata.json file
-            var metadata = new ArchiveMetadata(CurrentMetadataVersion, entrypointPath, files, CurrentBicepVersion, UriDocumentLinksToPathBasedLinks(baseFolderUri, documentLinks));
+            var metadata = new ArchiveMetadata(CurrentMetadataVersion, CurrentBicepVersion, entrypointPath, files, documentLinks);
             return JsonSerializer.Serialize(metadata, MetadataSerializationContext.Default.ArchiveMetadata);
         }
 
         private static IReadOnlyDictionary<string, SourceCodeDocumentPathLink[]>? UriDocumentLinksToPathBasedLinks(
-            Uri baseFolderUri,
+            IReadOnlyDictionary<Uri, string> sourceUriToRelativePathMap,
             IReadOnlyDictionary<Uri, SourceCodeDocumentUriLink[]>? uriBasedDocumentLinks
         )
         {
             return uriBasedDocumentLinks?.Select(
                 x => new KeyValuePair<string, SourceCodeDocumentPathLink[]>(
-                    CalculateRelativeFilePath(baseFolderUri, x.Key),
-                    x.Value.Select(link => DocumentPathLinkFromUriLink(baseFolderUri, link)).ToArray()
+                    sourceUriToRelativePathMap[x.Key],
+                    x.Value.Select(link => DocumentPathLinkFromUriLink(sourceUriToRelativePathMap, link)).ToArray()
                 )).ToImmutableDictionary();
         }
 
-        private static SourceCodeDocumentPathLink DocumentPathLinkFromUriLink(Uri baseFolderUri, SourceCodeDocumentUriLink uriBasedLink)
+        private static SourceCodeDocumentPathLink DocumentPathLinkFromUriLink(
+            IReadOnlyDictionary<Uri, string> sourceUriToRelativePathMap,
+            SourceCodeDocumentUriLink uriBasedLink)
         {
             return new SourceCodeDocumentPathLink(
                 uriBasedLink.Range,
-                CalculateRelativeFilePath(baseFolderUri, uriBasedLink.Target));
+                sourceUriToRelativePathMap[uriBasedLink.Target]);
         }
 
         private static void WriteNewFileEntry(TarWriter tarWriter, string archivePath, string contents)
         {
+            Debug.Assert(!archivePath.Contains('\\'), $"Source archive paths should not contain backslashes, only forward slashes: \"{archivePath}\"");
+            Debug.Assert(!archivePath.Contains(':'), $"Source archive paths should not contain drive letters or colons: \"{archivePath}\"");
+            Debug.Assert(!archivePath.Contains("/../") && !archivePath.StartsWith("../"), $"Source archive paths should not contain \"..\" folders: \"{archivePath}\"");
```

Comment:
That's a very good catch - should probably also make sure @asilverman is aware of this for type publishing.

Now I'm wondering if I could publish a types package containing `../../../C:/Windows/system32/notavirus.dll`

Reply:
He has been notified. :-)